### PR TITLE
Users table

### PR DESCRIPTION
**Pull request by EdiOancea**

### CODE REVIEW

**Comment by EdiOancea** [comment]
[Create users table and model](https://trello.com/c/OuZdrkbP/2-create-users-table-and-model)

File: app.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+const express = require('express');
+const Sequelize = require('sequelize');
```
Comment (lgnstn) [inline app.js]: If you keep adding everything to this file, you will end up with a few thousand lines of code here. I don't think you want this to happen :)
Let's rename this file `index.js` and keep here only the initial content, from the previous PR.
Then create a folder `app` and `app/models`. In `modes` you will add all the models, a separate file for each one, in this case `app/models/user.js`.

File: app.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+const app = express();
+const connection = new Sequelize({
```
Comment (lgnstn) [inline app.js]: Please move this connection into `app/models/index.js`. And then I'll show you how you can load all the models automatically.

File: app.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  username: 'ferret',
+  password: null,
+  host: 'localhost',
+  dialect: 'postgres',
+  define: {
+    timestamps: false,
+  },
+});
+
+const User = connection.define('user', {
```
Comment (lgnstn) [inline app.js]: As I said, this goes into `app/models/user.js`.

Comment (lgnstn) [reply app.js]: I know I didn't say anything about this, but we are going to use DB migrations to handle database changes :)

File: app.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    type: Sequelize.STRING,
+    field: 'last_name',
+  },
+  email: {
+    type: Sequelize.STRING,
+    field: 'email',
+  },
+  password: {
+    type: Sequelize.STRING,
+    field: 'email',
```
Comment (lgnstn) [inline app.js]: You sure?

**Review by lgnstn — CHANGES_REQUESTED** [review]

**Comment by EdiOancea** [comment]
@lgnstn updated

File: app/models/user.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+module.exports = (sequelize, DataTypes) => {
+  const User = sequelize.define('user', {
+    firstName: {
+      type: Sequelize.STRING,
+      field: 'first_name',
+      unique: 'compositeIndex',
+      validate: {
+        len: [3, 20],
+        firstNameValidator: (value) => {
```
Comment (lgnstn) [inline app/models/user.js]: There should be something like pattern validator. This is too much/complicated.

File: app/models/user.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+      allowNull: false,
+    },
+    active: {
+      type: Sequelize.BOOLEAN,
+      field: 'active',
+      defaultValue: true,
+      allowNull: false,
+    },
+  }, {});
+  User.associate = function(models) {
```
Comment (lgnstn) [inline app/models/user.js]: Please remove it if it does nothing.

File: app/migrations/20190319143718-create-user.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        field: 'email',
+      },
+      password: {
+        type: Sequelize.STRING,
+        field: 'password',
+      },
+      rating: {
+        type: Sequelize.DOUBLE,
+        field: 'rating',
+        min: 0,
```
Comment (lgnstn) [inline app/migrations/20190319143718-create-user.js]: Please remove these.

**Review by lgnstn — CHANGES_REQUESTED** [review]

**Comment by EdiOancea** [comment]
@lgnstn updated

**Comment by lgnstn** [comment]
@EdiOancea Please resolve the conflicts. Thanks!

**Review by lgnstn — APPROVED** [review]

**Pull request merged**